io-engine/src/core/volume_snapshot.rs

```diff
@@ -0,0 +1,46 @@
+use super::CoreError;
+use crate::{
+    lvs::{Lvol, LvolSnapshot},
+    subsys::NvmfReq,
+};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_bdev_io;
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait(?Send)]
+pub trait VolumeSnapshotOps {
+    /// Create Snapshot Common API.
+    async fn create_snapshot(&self) -> Result<u64, CoreError>;
+
+    /// Return the list of Snapshots for Local/remote Device.
+    fn list_snapshot(&self, source_vol: &str) -> Vec<LvolSnapshot>;
+}
+pub trait VolumeSnapshotParams {
```

Comment:
nit: Do we want all of these prefixed by Volume? Could this cause some confusion since they're not really a mayastor volume as we know it.